test_runner/regress/test_remote_storage.py

```diff
@@ -118,6 +114,14 @@ def test_remote_storage_backup_and_restore(
         wait_for_upload(client, tenant_id, timeline_id, current_lsn)
         log.info(f"upload of checkpoint {checkpoint_number} is done")
 
+    # Check that we had to retry the uploads
```

Comment:
We may want to add a UUID to each operation result, allowing us to use it as breadcrumbs in investigations and in tests as well, for example, the log_cotains will look for a specific upload operation

Reply:
:+1: would very much be fan of deconstructing all outcomes into internally pubsubbed events with correlation and causation id linkage to control plane requests or other inputs to the system (timed actions) -- hoping to prototype something along these lines next year :) I think it's quite a large jump however from current setup.

Reply:
Also, we know how many retries exactly we had to do, might assert that too.

Reply:
Created a new issue to track that idea, it seems out of scope for this PR